### PR TITLE
sol for moo route + other minor fixes

### PR DESCRIPTION
**Pull request by devo1ution**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: content/1_General/Adding_Solution.mdx [hunk 1]
```diff
@@ -118,7 +118,9 @@ See [Working With MDX](/general/working-mdx) for additional information.
       "isStarred": false,
       "tags": ["Binary Search", "Sorted Set"],
       "solutionMetadata": {
-        "kind": "internal"
+        "kind": "internal",
+        // "hasHint": true
```
Comment (bqi343) [inline content/1_General/Adding_Solution.mdx:122]: from #3740 this should be `hasHints`, not `hasHint`

**Review by bqi343 — APPROVED** [review]
the files I saw looked fine. didn't check the moo route solution

File: solutions/gold/usaco-1283.mdx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+#include <iostream>
+using namespace std;
+using ll = long long;
+
+const int A = 1e6 + 1;
+const int M = 1e9 + 7;
+const int N = 1e5;
+
+int n;
+ll F[A], I[A], a[N];
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:70]: better names

File: solutions/gold/usaco-1283.mdx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+// binary exponentiation:
+// https://usaco.guide/gold/modular#modular-exponentiation
+ll P(ll x, ll y) {
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:74]: can you just use the code in that module directly without any modifications?
helps keep things consistent

File: solutions/gold/usaco-1283.mdx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	ll r = 1;
+	for (y <<= 1; y >>= 1; x = x * x % M)
+		if (y & 1) r = r * x % M;
+	return r;
+}
+
+// n choose k = n!/[(n - k)!k!]
+ll C(ll n, ll k) { return F[n] * I[n - k] % M * I[k] % M; }
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:82]: i believe there is a combo module

File: solutions/gold/usaco-1283.mdx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	}
+	
+	cin >> n;
+	for (int i = 0; i < n; i++) {
+		cin >> a[i];
+		// convert a[i] to b[i]
+		a[i] /= 2;
+	}
+
+	ll r = 1;
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:101]: better name

File: solutions/gold/usaco-1235.mdx [hunk 6]
```diff
@@ -4,6 +4,7 @@ source: USACO Gold 2022 Open
 title: Balancing a Tree
 author: Daniel Zhu
 ---
+[Official Analysis (C++/Java)](http://www.usaco.org/current/data/sol_prob3_gold_open22.html)
```
Comment (ryanchou-dev) [inline solutions/gold/usaco-1235.mdx:7]:
```suggestion
[Official Analysis (C++ and Java)](http://www.usaco.org/current/data/sol_prob3_gold_open22.html)
```

File: content/4_Gold/Combinatorics.problems.json [hunk 7]
```diff
@@ -137,8 +137,8 @@
       "isStarred": false,
       "tags": ["Combinatorics"],
       "solutionMetadata": {
-        "kind": "USACO",
-        "usacoId": "1283"
+        "kind": "internal",
+        "hasHints": true
```
Comment (ryanchou-dev) [inline content/4_Gold/Combinatorics.problems.json:141]: hm? when did we add this field 👀

File: solutions/gold/usaco-1283.mdx (hunk 5 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-1283.mdx:101]: also why does it start @ 1

File: solutions/gold/usaco-1283.mdx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+	for (y <<= 1; y >>= 1; x = x * x % M)
+		if (y & 1) r = r * x % M;
+	return r;
+}
+
+// n choose k = n!/[(n - k)!k!]
+ll C(ll n, ll k) { return F[n] * I[n - k] % M * I[k] % M; }
+
+signed main() {
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:84]:
```suggestion
int main() {
```

File: solutions/gold/usaco-1283.mdx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+	cin.tie(0)->sync_with_stdio(0);
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:85]: remove if runs w/o

**Review by ryanchou-dev — APPROVED** [review]
AMAZINGB

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/4_Gold/Combinatorics.problems.json (hunk 7 above)
Comment (devo1ution) [reply content/4_Gold/Combinatorics.problems.json:141]: a while back lol in like #3740 but no one used it :(

File: solutions/gold/usaco-1283.mdx (hunk 4 above)
Comment (devo1ution) [reply solutions/gold/usaco-1283.mdx:82]: yeah thats the module this problem is in i just added the formula there for context

File: content/4_Gold/Combinatorics.problems.json (hunk 7 above)
Comment (SansPapyrus683) [reply content/4_Gold/Combinatorics.problems.json:141]: haha i'm so sorry daniel

File: solutions/gold/usaco-1283.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/gold/usaco-1283.mdx:82]: if that's the case link + use the code there

File: solutions/gold/usaco-1283.mdx [hunk 10]
```diff
@@ -67,28 +67,31 @@ const int M = 1e9 + 7;
...
+ll fact[A], inv_fact[A], a[N];
 
 // binary exponentiation:
 // https://usaco.guide/gold/modular#modular-exponentiation
-ll P(ll x, ll y) {
-	ll r = 1;
-	for (y <<= 1; y >>= 1; x = x * x % M)
-		if (y & 1) r = r * x % M;
-	return r;
+ll P(ll x, ll n) {
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:74]: chance function name asw

File: solutions/gold/usaco-1283.mdx [hunk 11]
```diff
@@ -98,16 +101,17 @@ signed main() {
 		a[i] /= 2;
 	}
 
-	ll r = 1;
+	ll ans = 1;  // C_{n - 1} -> 1 valid path
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:104]: what does the comment mean

File: solutions/gold/usaco-1283.mdx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+```cpp
+#include <iostream>
+using namespace std;
+using ll = long long;
+
+const int A = 1e6 + 1;
+const int M = 1e9 + 7;
+const int N = 1e5;
+
+int n;
```
Comment (SansPapyrus683) [inline solutions/gold/usaco-1283.mdx:69]: does this have to be global

**Review by SansPapyrus683 — APPROVED** [review]
the bad thing about being a pm is that you have to review solutions from people that are orders of magnitude more intelligent than you are

**Pull request merged**